src/hotspot/cpu/x86/x86.ad

```diff
@@ -1807,9 +1807,15 @@ const bool Matcher::match_rule_supported_vector(int opcode, int vlen, BasicType
     case Op_LoadVectorGatherMasked:
     case Op_StoreVectorScatterMasked:
     case Op_StoreVectorScatter:
-      if(is_subword_type(bt)) {
+      if (is_subword_type(bt)) {
         return false;
-      } else if (size_in_bits < 512 && !VM_Version::supports_avx512vl()) {
+      }
+      // fallthrough
+    case Op_CompressV:
+      if (is_subword_type(bt) && !VM_Version::supports_avx512_vbmi2()) {
+        return false;
+      }
+      if (size_in_bits < 512 && !VM_Version::supports_avx512vl()) {
```

Comment:
It would be better to add a separate case for Op_CompressV instead of adding it as a fallthrough to gather/scatter.